src/renderers/dom/shared/dangerousStyleValue.js

```diff
@@ -48,7 +51,27 @@ function dangerousStyleValue(name, value) {
   }
 
   if (typeof value === 'string') {
-    value = value.trim();
+    if (__DEV__) {
+      if (component) {
+        var owner = component._currentElement._owner;
+        var componentName = owner ? owner.getName() : component._currentElement.type;
+        if (!styleWarnings[componentName]) {
+          styleWarnings[componentName] = {};
+        }
+        var warnings = styleWarnings[componentName];
+        if (!warnings[name]) {
+          warnings[name] = true;
+          warning(
+            false,
+            '`%s` passed a numeric string value for CSS property `%s` ' +
+            'which will be treated as a unitless number.',
+            componentName,
+            name
+          );
+        }
+      }
+    }
+    return value.trim();
   }
   return value + 'px';
```

Comment:
That behaviour hasn't actually changed. Otherwise the tests would be falling, anyway. If you look at my first commit you'll see that the only actual change in logic is that the reassignment has been replaced with a `return` statement. This only affects *unitless* numeric string values.